src/Tools/dotnet-counters/CounterMonitor.cs

```diff
@@ -35,7 +39,7 @@ public CounterMonitor()
             pauseCmdSet = false;
         }
 
-        private void Dynamic_All(TraceEvent obj)
+        private void DynamicAllMonitor(TraceEvent obj)
```

Comment:
I think we could refactor this to have less duplication. The ConsoleWriter and ICounterExporter aren't that different from an interface perspective. Both of them are components whose job is to receive counter events, render those values somewhere (in the console or in a file format), and respond to updates in program state. If you abstracted ICounterExporter a little more both component could fit behind the same interface and a bunch of code here would collapse. I didn't go through all the details but maybe something similar to this?

interface ICounterRenderer
{
   Initizalize();
   EventPipeSourceConnected(); 
   ToggleStatus(paused);
   CounterPayloadReceived(providerName, payload, paused);
   Stop();
}

If you did then exporter.Write() and writer.Update would both become renderer.CounterPayloadReceived() and these two handlers could collapse into a single handler. The file exporter might get a ToggleStatus() call that it doesn't care about, but that can be implemented with an empty function. Given some common functionality that all the file export implementations would want to share a simple base class would probably also make sense.

Separate from the factoring we've also got the issue that the pause/resume updates only occur as fast as dotnet-counters receives event updates. As a future step you might want to decouple ToggleStatus from the event updates.